src/JsonSerializer/JsonSerializer.php

```diff
@@ -231,6 +293,8 @@ protected function extractObjectData($value, $ref, $properties)
      */
     protected function unserializeData($value)
     {
+        $value = $this->decodeNonUtf8FromUtf8($value);
```

Comment:
@ghola To avoid looping thru the whole data, which can be very expensive if the data is big, you can check with `strpos` if the 2 new constant values are in the code before executing this method.